streams/src/test/java/org/apache/kafka/streams/integration/RocksDBMetricsIntegrationTest.java

```diff
@@ -142,21 +155,58 @@ public void shouldExposeRocksDBMetricsForSegmentedStateStoreBeforeAndAfterFailur
         final Properties streamsConfiguration = streamsConfig();
         IntegrationTestUtils.purgeLocalStreamsState(streamsConfiguration);
         final StreamsBuilder builder = builderForSegmentedStateStore();
+        final String metricsScope = "rocksdb-window-state-id";
+
+        cleanUpStateRunAndVerify(
+            builder,
+            streamsConfiguration,
+            LongDeserializer.class,
+            LongDeserializer.class,
+            this::verifyThatRocksDBMetricsAreExposed,
+            metricsScope
+        );
 
         cleanUpStateRunAndVerify(
             builder,
             streamsConfiguration,
             LongDeserializer.class,
             LongDeserializer.class,
-            "rocksdb-window-state-id"
+            this::verifyThatRocksDBMetricsAreExposed,
+            metricsScope
+        );
```

Comment:
Do I read this right, that we're doing the exact same thing twice?

Reply:
As the test name says `shouldExposeRocksDBMetricsForSegmentedStateStoreBeforeAndAfterFailureWithEmptyStateDir()` this test verifies that the metrics are still there after a failure and an empty state dir (see https://issues.apache.org/jira/browse/KAFKA-9355). 
I basically simulate a failure by closing the app, wiping out the state, re-starting the app, verifying the exposure of the metrics, and closing the app. That results in doing twice the same thing.

Reply:
Oh, I see. Definitely wasn't obvious by reading the test.

Reply:
I tried to make the tests clearer.